opm/simulators/flow/Main.hpp

```diff
@@ -325,77 +326,89 @@ class Main
 
 #if HAVE_DAMARIS
         enableDamarisOutput_ = EWOMS_GET_PARAM(PreTypeTag, bool, EnableDamarisOutput);
+        // Reset to false as we cannot use Damaris if there is only one rank.
+        if ((enableDamarisOutput_ == true) && (EclGenericVanguard::comm().size() == 1)) {
+            std::string msg ;
+            msg = "\nUse of Damaris (command line argument --enable-damaris-output=true) has been dissabled for run with only one rank.\n" ;
+            OpmLog::info(msg);
+            enableDamarisOutput_ = false ;
+        }
+        
         if (enableDamarisOutput_) {
             this->setupDamaris(outputDir,
                                EWOMS_GET_PARAM(PreTypeTag, bool, EnableDamarisOutputCollective));
         }
 #endif // HAVE_DAMARIS
 
-        int mpiRank = EclGenericVanguard::comm().rank();
-        outputCout_ = false;
-        if (mpiRank == 0)
-            outputCout_ = EWOMS_GET_PARAM(PreTypeTag, bool, EnableTerminalOutput);
+        // Guard for when the Damaris core(s) return from damaris_start() 
+        // which happens when damaris_stop() is called in main simulation
+        if (isSimulationRank_) {
```

Comment:
Doing it in this way indents all the existing code, without changing it. I think it would be better to write
```
if (isSimulationRank_) {
    exitCode = EXIT_SUCCESS;
    return true;
}
```
and then leave the rest of the function unchanged.

Reply:
I do not find that as clear, however functionality is the same and it is good to keep indentation under control, so done.